specification.md

```diff
@@ -896,11 +931,28 @@ The AgentDescription message has the following structure:
 
 ```protobuf
 message AgentDescription {
-    repeated KeyValue identifying_attributes = 1;
-    repeated KeyValue non_identifying_attributes = 2;
+    bytes hash = 1;
+    repeated KeyValue identifying_attributes = 2;
+    repeated KeyValue non_identifying_attributes = 3;
 }
 ```
 
+#### hash
+
+The hash of the content of all other fields (even if the other fields are omitted
+for compression).
+
+If the content of the other fields did not change since it was last reported the Agent is
```

Comment:
Maybe the general hash explanation could be put in a "Agent To Server Synchronization" subsection and the details on it could be referenced here and in other places where hash is present (rather than refer to Agent Description hash)? I think that would make for a more clear structure

Reply:
Re-worded. PTAL.